src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -320,11 +322,17 @@ struct PhotonCreator
     DINLINE void operator()(Electron& electron, Photon& photon) const
     {
         photon[multiMask_] = 1;
-        photon[localCellIdx_] = electron[localCellIdx_];
-        photon[position_] = electron[position_];
         photon[momentum_] = this->photon_mom;
         electron[momentum_] -= this->photon_mom;
-        photon[weighting_] = electron[weighting_];
+        PMACC_AUTO(destPhoton,
+            deselect<
+                boost::mpl::vector<
+                    multiMask,
+                    momentum
+                >
+            >(photon)
+        );
+        particles::operations::assign( destPhoton, electron );
```

Comment:
I think it would be saver to first copy the electron attributes to the photons (with `deselect` if you want to), and after that write the new attributes just in case someone alters your code and forgets to include the attribute set in the `deselect` list.